src/Tokenizer.php

```diff
@@ -100,61 +95,73 @@ protected function statement()
 
                 $this->delimiter_pattern = preg_quote($delimiter);
 
-                continue; // omits DELIMITER command - it isn't part of SQL statement syntax
+            } else {
+                $tokens[] = $token;
             }
-
-            $tokens[] = $token;
+            $token = $this->token();
         }
 
         return $tokens;
     }
 
     /**
+     * TODO: Refactor this - cyclomatic complexity > 10
```

Comment:
Do we have a task on this?

Reply:
I've added a jira ticket. I will send you a link via Teams :+1: